tests/unit/test__upload.py

```diff
@@ -621,12 +621,11 @@ def test__prepare_request_success_with_headers(self):
         new_headers = self._prepare_request_helper(headers)
         assert new_headers is not headers
         expected_headers = {
+            "cannot": "touch this",
```

Comment:
This test is very subtle and relies on https://github.com/googleapis/google-resumable-media-python/blob/cee416449701b72e7fd532585a1f739b02b6ab32/tests/unit/test__upload.py#L581 to pass in new headers when calling `self._upload_in_flight(data, headers=headers, checksum=checksum)`. IIUC, the test is to prevent any regressions from future changes (super not clear by the name of the test).

What I don't understand is what changed when `**self._headers` is added so it can be tested. @Breathtender could add a test that validates the behavior change intended by adding `**self._headers`?

Reply:
Oh that's a good change lol